lib/Echidna/Solidity.hs

```diff
@@ -126,22 +127,27 @@ contracts fp = let usual = ["--solc-disable-warnings", "--export-format", "solc"
     let solargs = a ++ linkLibraries ls & (usual ++) .
                   (\sa -> if null sa then [] else ["--solc-args", sa])
         fps = toList fp
-        compileOne :: (MonadIO m, MonadThrow m, MonadReader x m, Has SolConf x) => FilePath -> m [SolcContract]
+        compileOne :: (MonadIO m, MonadThrow m, MonadReader x m, Has SolConf x) => FilePath -> m ([SolcContract], SourceCache)
         compileOne x = do
           mSolc <- liftIO $ do
             stderr <- if q then UseHandle <$> openFile "/dev/null" WriteMode else pure Inherit
             (ec, out, err) <- readCreateProcessWithExitCode (proc path $ (c ++ solargs) |> x) {std_err = stderr} ""
             case ec of
               ExitSuccess -> readSolc "crytic-export/combined_solc.json"
               ExitFailure _ -> throwM $ CompileFailure out err
-          maybe (throwM SolcReadFailure) (pure . toList . fst) mSolc
-    concat <$> sequence (compileOne <$> fps)
+            
+          maybe (throwM SolcReadFailure) (pure . first toList) mSolc
+    cps <- mapM compileOne fps
+    let (cs, ss) = unzip cps
+    let lss = length ss
+    when (lss > 1) $ liftIO $ putStrLn "WARNING: more than one SourceCache was found after compile. Only the first one will be used."
+    return (concat cs, head ss)
```

Comment:
`return` -> `pure`, there is partial `head` maybe we could `throwM` instead?

Reply:
> return -> pure,

Fixed.

>  there is partial head maybe we could throwM instead?


I believe `ss` cannot be empty, otherwise the result of the compilation will be "invalid".